Spring `@RequestMapping` doesn't work with `quarkus-resteasy-reactive-qute`

### Describe the bug

Quarkus version: upstream

`quarkus-spring-web` is not working anymore with `quarkus-resteasy-qute` due to dependency conflicts, so I have to move on to `quarkus-resteasy-reactive-qute`,  but this library doesn't support Spring `@RequestMapping` annotation without method (Is supported in previous versions).

Extensions involved:
```
<dependency>
            <groupId>io.quarkus</groupId>
            <artifactId>quarkus-spring-web</artifactId>
 </dependency>

<dependency>
            <groupId>io.quarkus</groupId>
            <artifactId>quarkus-resteasy-reactive-qute</artifactId>
</dependency>
``` 

Code example:

```
@RestController
public class SimpleController {

    @Value("${spring.application.name}")
    String appName;

    @Inject
    Template home;

    @RequestMapping(value = "/", produces = MediaType.TEXT_HTML)
    public TemplateInstance homePage() {
        return home.data("appName", appName);
    }
}
```
Stacktrace: 
```
[ERROR] Failed to execute goal io.quarkus:quarkus-maven-plugin:999-SNAPSHOT:build (build) on project spring-web: Failed to build quarkus application: io.quarkus.builder.BuildException: Build failure: Build failed due to errors
[ERROR]         [error]: Build step io.quarkus.resteasy.reactive.server.deployment.ResteasyReactiveProcessor#setupEndpoints threw an exception: java.lang.RuntimeException: java.lang.IllegalArgumentException: Usage of '@RequestMapping' without an http method is not allowed. Offending method is 'io.quarkus.ts.spring.web.boostrap.web.SimpleController#homePage'
[ERROR]         at org.jboss.resteasy.reactive.common.processor.EndpointIndexer.createEndpoints(EndpointIndexer.java:256)
[ERROR]         at io.quarkus.resteasy.reactive.server.deployment.ResteasyReactiveProcessor.setupEndpoints(ResteasyReactiveProcessor.java:449)
[ERROR]         at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
[ERROR]         at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
[ERROR]         at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
[ERROR]         at java.base/java.lang.reflect.Method.invoke(Method.java:566)
[ERROR]         at io.quarkus.deployment.ExtensionLoader$2.execute(ExtensionLoader.java:820)
[ERROR]         at io.quarkus.builder.BuildContext.run(BuildContext.java:277)
[ERROR]         at org.jboss.threads.ContextHandler$1.runWith(ContextHandler.java:18)
[ERROR]         at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2449)
[ERROR]         at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1478)
[ERROR]         at java.base/java.lang.Thread.run(Thread.java:829)
[ERROR]         at org.jboss.threads.JBossThread.run(JBossThread.java:501)
[ERROR] Caused by: java.lang.IllegalArgumentException: Usage of '@RequestMapping' without an http method is not allowed. Offending method is 'io.quarkus.ts.spring.web.boostrap.web.SimpleController#homePage'
[ERROR]         at io.quarkus.spring.web.deployment.SpringWebProcessor$1.transform(SpringWebProcessor.java:174)
[ERROR]         at org.jboss.resteasy.reactive.common.processor.transformation.AnnotationStore.transform(AnnotationStore.java:98)
[ERROR]         at java.base/java.util.concurrent.ConcurrentHashMap.computeIfAbsent(ConcurrentHashMap.java:1737)
[ERROR]         at org.jboss.resteasy.reactive.common.processor.transformation.AnnotationStore.getAnnotations(AnnotationStore.java:51)
[ERROR]         at org.jboss.resteasy.reactive.common.processor.transformation.AnnotationStore.getAnnotation(AnnotationStore.java:64)
[ERROR]         at org.jboss.resteasy.reactive.common.processor.EndpointIndexer.createEndpoints(EndpointIndexer.java:286)
[ERROR]         at org.jboss.resteasy.reactive.common.processor.EndpointIndexer.createEndpoints(EndpointIndexer.java:227)
[ERROR]         ... 12 more


```

The workaround is to use the annotation `@GetMapping(value = "/", produces = MediaType.TEXT_HTML)` or `@RequestMapping(value = "/", produces = MediaType.TEXT_HTML, method = RequestMethod.GET)`

This breaking change should be documented.

--- a/extensions/spring-web/deployment/src/main/java/io/quarkus/spring/web/deployment/SpringWebProcessor.java
+++ b/extensions/spring-web/deployment/src/main/java/io/quarkus/spring/web/deployment/SpringWebProcessor.java
@@ -178,23 +178,22 @@ public class SpringWebProcessor {
                 if (mappingAnnotationInstance != null) {
                     AnnotationValue methodValue = mappingAnnotationInstance.value("method");
                     if (methodValue == null) {
-                        throw new IllegalArgumentException(
-                                "Usage of '@RequestMapping' without an http method is not allowed. Offending method is '"
-                                        + methodInfo.declaringClass().name() + "#" + methodInfo.name() + "'");
+                        jaxRSMethodAnnotation = ResteasyReactiveDotNames.GET;
+                    } else {
+                        String[] methods = methodValue.asEnumArray();
+                        if (methods.length > 1) {
+                            throw new IllegalArgumentException(
+                                    "Usage of multiple methods using '@RequestMapping' is not allowed. Offending method is '"
+                                            + methodInfo.declaringClass().name() + "#" + methodInfo.name() + "'");
+                        }
+                        DotName methodDotName = ResteasyReactiveScanner.METHOD_TO_BUILTIN_HTTP_ANNOTATIONS.get(methods[0]);
+                        if (methodDotName == null) {
+                            throw new IllegalArgumentException(
+                                    "Unsupported HTTP method '" + methods[0] + "' for @RequestMapping. Offending method is '"
+                                            + methodInfo.declaringClass().name() + "#" + methodInfo.name() + "'");
+                        }
+                        jaxRSMethodAnnotation = methodDotName;
                     }
-                    String[] methods = methodValue.asEnumArray();
-                    if (methods.length > 1) {
-                        throw new IllegalArgumentException(
-                                "Usage of multiple methods using '@RequestMapping' is not allowed. Offending method is '"
-                                        + methodInfo.declaringClass().name() + "#" + methodInfo.name() + "'");
-                    }
-                    DotName methodDotName = ResteasyReactiveScanner.METHOD_TO_BUILTIN_HTTP_ANNOTATIONS.get(methods[0]);
-                    if (methodDotName == null) {
-                        throw new IllegalArgumentException(
-                                "Unsupported HTTP method '" + methods[0] + "' for @RequestMapping. Offending method is '"
-                                        + methodInfo.declaringClass().name() + "#" + methodInfo.name() + "'");
-                    }
-                    jaxRSMethodAnnotation = methodDotName;
                 } else {
                     if (methodInfo.hasAnnotation(GET_MAPPING)) {
                         jaxRSMethodAnnotation = ResteasyReactiveDotNames.GET;

--- a/extensions/spring-web/deployment/src/test/java/io/quarkus/spring/web/test/basic/BasicMappingTest.java
+++ b/extensions/spring-web/deployment/src/test/java/io/quarkus/spring/web/test/basic/BasicMappingTest.java
@@ -29,6 +29,15 @@ public class BasicMappingTest {
     }
 
     @Test
+    public void verifyRequestMappingWithNoMethod() {
+        when().get(TestController.CONTROLLER_PATH + "/hello4?name=people")
+                .then()
+                .statusCode(200)
+                .contentType("text/plain")
+                .body(is("hello people"));
+    }
+
+    @Test
     public void verifyGetToMethodWithoutForwardSlash() {
         when().get(TestController.CONTROLLER_PATH + "/yolo")
                 .then()

--- a/extensions/spring-web/deployment/src/test/java/io/quarkus/spring/web/test/basic/TestController.java
+++ b/extensions/spring-web/deployment/src/test/java/io/quarkus/spring/web/test/basic/TestController.java
@@ -23,6 +23,11 @@ public class TestController {
         return "hello " + name;
     }
 
+    @RequestMapping("/hello4")
+    public String requestMappingNoMethod(@RequestParam String name) {
+        return "hello " + name;
+    }
+
     @GetMapping("yolo")
     public String yolo() {
         return "yolo";

